server/application-server/src/main/java/de/tum/in/www1/hephaestus/chat/message/MessageService.java

```diff
@@ -0,0 +1,84 @@
+package de.tum.in.www1.hephaestus.chat.message;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import de.tum.in.www1.hephaestus.chat.message.Message.MessageSender;
+import de.tum.in.www1.hephaestus.chat.session.Session;
+import de.tum.in.www1.hephaestus.chat.session.SessionRepository;
+import de.tum.in.www1.hephaestus.intelligenceservice.ApiClient;
+import de.tum.in.www1.hephaestus.intelligenceservice.api.DefaultApi;
+import de.tum.in.www1.hephaestus.intelligenceservice.model.ChatRequest;
+import de.tum.in.www1.hephaestus.intelligenceservice.model.ChatResponse;
+
+@Service
+public class MessageService {
+
+    private DefaultApi sessionApiClient;
+    @Autowired
+    private SessionRepository sessionRepository;
+    @Autowired
+    private MessageRepository messageRepository;
+
+    public MessageService() {
+        ApiClient apiClient = new ApiClient();
+        apiClient.setBasePath("http://127.0.0.1:8000");
+        this.sessionApiClient = new DefaultApi(apiClient);
+    }
+
+    public List<MessageDTO> getMessagesBySessionId(Long sessionId) {
+        return messageRepository.findBySessionId(sessionId).stream()
+                .map(message -> new MessageDTO(message.getId(), message.getSentAt(), message.getSender(),
+                        message.getContent(), message.getSession().getId()))
+                .toList();
+    }
+
+    public MessageDTO sendMessage(String content, Long sessionId) {
+        Optional<Session> session = sessionRepository.findById(sessionId);
+        if (session.isEmpty() || content == null) {
+            return null;
+        }
+
+        Message userMessage = new Message();
+        userMessage.setSender(MessageSender.USER);
+        userMessage.setContent(content);
+        userMessage.setSession(session.get());
```

Comment:
Maybe just create a new variable for the unpacked session before accessing it via the `Optional` that often